### PR TITLE
Pageskin ads targeted by series tag

### PR DESCRIPTION
**Pull request by jeteve**
## What does this change?

This adds PageSkin targetting by "series" type of tag per card:

https://trello.com/c/VUOsXQti/64-add-series-tag-targeting-for-page-skins

### CODE REVIEW

File: common/app/common/dfp/PageSkinSponsorship.scala [hunk 1]
```diff
@@ -3,15 +3,22 @@ package common.dfp
...
+  *   because DFP doesn't have this concept.
+ */
 case class PageSkinSponsorship(lineItemName: String,
                                lineItemId: Long,
                                adUnits: Seq[String],
+                              // Targeting properties
                                editions: Seq[Edition],
                                countries: Seq[String],
-                               isR2Only: Boolean,
+                               isR2Only: Boolean, // Legacy platform.
```
Comment (kelvin-chappell) [inline common/app/common/dfp/PageSkinSponsorship.scala]: This is redundant now.  But probably to be removed in a separate PR.

File: .gitignore [hunk 2]
```diff
@@ -51,6 +51,12 @@ static/riffraff
 
 static/src/stylesheets/icons/*
 
+# cse ( https://github.com/jeteve/App-CSE/ ) #
```
Comment (LATaylor-guardian) [inline .gitignore]: I'm not sure this should have been committed?

Comment (LATaylor-guardian) [reply .gitignore]: Looks like a change specific to you and could be applied to your global `.gitignore`.

File: common/app/common/dfp/DfpData.scala [hunk 3]
```diff
@@ -148,13 +149,13 @@ case class GuTargeting(adUnitsIncluded: Seq[GuAdUnit],
...
                        customTargetSets: Seq[CustomTargetSet]) {
 
-  val keyValues: Seq[String] = {
-    for {
-      targetSet <- customTargetSets
-      target <- targetSet.targets if target.isKeywordTag
-      targetValue <- target.values
-    } yield targetValue
-  }
+  // This is local pure helper function:
```
Comment (LATaylor-guardian) [inline common/app/common/dfp/DfpData.scala]: This comment is unnecessary and can be be removed.

File: common/app/common/dfp/DfpData.scala [hunk 4]
```diff
@@ -148,13 +149,13 @@ case class GuTargeting(adUnitsIncluded: Seq[GuAdUnit],
...
+  private val tagValues = ( ( customTargetSets: Seq[CustomTargetSet],
+                              isGoodTag: CustomTarget => Boolean )
```
Comment (LATaylor-guardian) [inline common/app/common/dfp/DfpData.scala]: Maybe a better name for `isGoodTag` is `tagFilter` because this function can literally filter the tags in a non prescribed way.

File: common/app/common/dfp/PageskinAdAgent.scala [hunk 5]
```diff
@@ -25,11 +25,25 @@ trait PageskinAdAgent {
       } else Seq.empty
     } else {
       val targetingMap = toMap(metaData.commercial.map(_.adTargeting(edition)).getOrElse(Set.empty))
-      val keywordTargeting = targetingMap.get("k") match {
-        case Some(values: MultipleValues) => values.values.toSeq
-        case _ => Seq.empty
-      }
-      candidates filter { sponsorship => sponsorship.keywords.intersect(keywordTargeting).nonEmpty }
+
+      // Little DRY helper
```
Comment (LATaylor-guardian) [inline common/app/common/dfp/PageskinAdAgent.scala]: Comment can be removed.

File: common/app/common/dfp/DfpData.scala [hunk 6]
```diff
@@ -148,13 +149,13 @@ case class GuTargeting(adUnitsIncluded: Seq[GuAdUnit],
...
 
-  val keyValues: Seq[String] = {
-    for {
-      targetSet <- customTargetSets
-      target <- targetSet.targets if target.isKeywordTag
-      targetValue <- target.values
-    } yield targetValue
-  }
+  // This is local pure helper function:
+  private val tagValues = ( ( customTargetSets: Seq[CustomTargetSet],
```
Comment (JonNorman) [inline common/app/common/dfp/DfpData.scala]: Could you add a type to this?

Comment (JonNorman) [reply common/app/common/dfp/DfpData.scala]: Actually, it is already implicit in the definition - sorry it wasn't clear because of the way GitHub has inserted the comments 👍

File: common/app/common/dfp/PageskinAdAgent.scala [hunk 7]
```diff
@@ -25,11 +25,25 @@ trait PageskinAdAgent {
...
+      val targetingMapValues = ((map: Map[String, AdTargetParamValue], key: String) =>
```
Comment (LATaylor-guardian) [inline common/app/common/dfp/PageskinAdAgent.scala]: I think this is probably an over abstraction. You can simply do:

```
val keywordTargeting = targetingMap.get("k").map(_.values).getOrElse(Seq.empty)
val seriesTargeting = targetingMap.get("se").map(_.values).getOrElse(Seq.empty)
```

File: common/app/common/dfp/PageskinAdAgent.scala [hunk 8]
```diff
@@ -25,11 +25,25 @@ trait PageskinAdAgent {
...
+        map.get(key) match {
+          case Some(values: MultipleValues) => values.values.toSeq
+          case _ => Seq.empty
+        }
+      )
+
+      val keywordTargeting = targetingMapValues( targetingMap , "k" )
+      val serieTargeting = targetingMapValues( targetingMap , "se" )
```
Comment (LATaylor-guardian) [inline common/app/common/dfp/PageskinAdAgent.scala]: s/serieTargeting/seriesTargeting/

File: common/app/common/dfp/PageskinAdAgent.scala [hunk 9]
```diff
@@ -25,11 +25,25 @@ trait PageskinAdAgent {
...
+
+      candidates filter { sponsorship => {
```
Comment (LATaylor-guardian) [inline common/app/common/dfp/PageskinAdAgent.scala]: The `{` after `=>` is not needed.

File: common/app/common/dfp/PageskinAdAgent.scala [hunk 10]
```diff
@@ -25,11 +25,25 @@ trait PageskinAdAgent {
...
+                           // Two ways of matching:
```
Comment (LATaylor-guardian) [inline common/app/common/dfp/PageskinAdAgent.scala]: Comments aren't needed. If you need to explain predicates more assign to well named variables.

File: common/app/common/dfp/PageSkinSponsorship.scala (hunk 1 above)
Comment (jeteve) [reply common/app/common/dfp/PageSkinSponsorship.scala]: 👍 Filed a card to get rid of it: https://trello.com/c/EHgmnw1i/142-get-rid-of-isr2only-in-pageskin-sponsorship

File: .gitignore (hunk 2 above)
Comment (jeteve) [reply .gitignore]: global .gitignore? Cool!

File: common/app/common/dfp/DfpData.scala (hunk 6 above)
Comment (jeteve) [reply common/app/common/dfp/DfpData.scala]: Actually I was wondering about that. I would prefer to use the 4th form from https://docs.scala-lang.org/style/declarations.html#function-values but I couldn't get it to work.

File: common/app/common/dfp/PageskinAdAgent.scala (hunk 7 above)
Comment (jeteve) [reply common/app/common/dfp/PageskinAdAgent.scala]: Not sure it is equivalent to the case class match, as AdTargetParamValue is a trait that doesn't have `values`

File: common/app/common/dfp/PageskinAdAgent.scala (hunk 10 above)
Comment (jeteve) [reply common/app/common/dfp/PageskinAdAgent.scala]: 👌

**Comment by LATaylor-guardian** [comment]
👍
